src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts

```diff
@@ -218,6 +222,67 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 			this._lastUserData = e.key;
 			this._lastUserDataTimestamp = Date.now();
 		}));
+		this._register(this._terminal.onData(e => {
+			if (e === '\x03') {
+				// Ctrl+C
+				this._sendTelemetryInfo(true);
+				this._mostRecentAcceptedCompletion = undefined;
+			}
+		}));
+	}
+
+	private _sendTelemetryInfo(fromInterrupt?: boolean): void {
+		const commandLine = this._mostRecentPromptInputState?.value;
+		const label = this._mostRecentAcceptedCompletion?.label;
+		const kind = this._mostRecentAcceptedCompletion?.kindLabel;
+		if (label === undefined || commandLine === undefined || kind === undefined) {
+			return;
+		}
+
+		let outcome: CompletionOutcome;
+		if (fromInterrupt) {
+			outcome = CompletionOutcome.Interrupted;
+		} else if (commandLine.trim() && commandLine.endsWith(label)) {
+			outcome = CompletionOutcome.Accepted;
```

Comment:
The most correct thing here would be whether the completion didn't change from its original position, but `includes` instead of `endsWith` is good enough.